### PR TITLE
Adding nats micro example for Rust

### PR DESCRIPTION
**Pull request by autodidaddict**
Adds an example for micro/intro in Rust. The change to the rust Dockerfile is, for some odd reason, required in order to allow output to show up from running the Rust examples (thanks @Jarema)

### CODE REVIEW

File: examples/micro/intro/rust/main.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// In this sample, we'll use the Rust SDK to create and start a service, which automatically
+// participates in the service discovery, stats management, and ping operations. The service
+// we're going to build exposes two endpoints:
+// * min - returns the minimum value in an array of integers
```
Comment (bruth) [inline examples/micro/intro/rust/main.rs:4]: Add a line space to (hopefully) render the bullets..

```suggestion

// * min - returns the minimum value in an array of integers
```

**Review by bruth — APPROVED** [review]
One minor addition to properly render the bullets in the first comment block.

**Comment by bruth** [comment]
Ugh, the output is empty.. same weird side effect with other Rust examples.. until we figure this out, let's generate the recording locally and commit. The CI will use that if present rather than generate one.

**Comment by autodidaddict** [comment]
w00t! :partying_face: the bullets rendered properly.

**Comment by autodidaddict** [comment]
> Ugh, the output is empty.. same weird side effect with other Rust examples.. until we figure this out, let's generate the recording locally and commit. The CI will use that if present rather than generate one.

It seems to have ignored the manually created `output.txt`

**Comment by bruth** [comment]
Still no output 🙁

**Comment by bruth** [comment]
Ah needs the output.cast also

**Comment by bruth** [comment]
A condition that checks for one but not the other..

**Comment by Jarema** [comment]
I will take a deeper look tomorrow in that generation issue with Docker

**Comment by autodidaddict** [comment]
@Jarema @bruth I'd like to get this merged.. please let me know what I can do in order to get the CI to pass here.

**Comment by bruth** [comment]
@Jarema pushed a commit to the `fix-rust-output` branch a few days ago that you can merge in here to test if it fixes the issue.

**Comment by Jarema** [comment]
Ok. Will work on this one today.

**Comment by autodidaddict** [comment]
These look like Python failures, which is weird... https://github.com/ConnectEverything/nats-by-example/actions/runs/6615260740/job/17967034931?pr=144#step:8:2174

**Pull request merged**